lib/ansible/module_utils/iosxr.py

```diff
@@ -26,9 +26,27 @@
 # LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
 # USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #
+from ansible.module_utils.basic import env_fallback
+from ansible.module_utils.network_common import exec_command, to_list
 
 _DEVICE_CONFIGS = {}
 
+iosxr_argument_spec = {
+    'host': dict(),
+    'port': dict(type='int'),
+    'username': dict(fallback=(env_fallback, ['ANSIBLE_NET_USERNAME'])),
+    'password': dict(fallback=(env_fallback, ['ANSIBLE_NET_PASSWORD']), no_log=True),
+    'timeout': dict(type='int', default=10),
+    'provider': dict(type='dict')
+}
+
+def check_args(module, warnings):
+    provider = module.params['provider'] or {}
+    for key in iosxr_argument_spec:
+        if key != 'provider' and module.params[key]:
+            warnings.append('argument %s has been deprecated and will be '
```

Comment:
Not tested, though this may lead to a similar issue as https://github.com/ansible/ansible/issues/20391 due to `timeout` having a default value